fedimint-core/src/config.rs

```diff
@@ -168,6 +168,10 @@ impl FederationId {
     fn try_from_bytes(bytes: [u8; 48]) -> Option<Self> {
         Some(Self(threshold_crypto::PublicKey::from_bytes(bytes).ok()?))
     }
+
+    pub fn to_pub_key(&self) -> Result<secp256k1::PublicKey, anyhow::Error> {
```

Comment:
@jkitman, @elsirion, would you have pointers on how would to derive a valid `secp256k1::PublicKey` from the threshold pubkey used in federation id?

Reply:
I'd do it like this:

```rust
    /// Converts a federation id to a public key to which we don't know the
    /// private key
    ///
    /// That is ok because we only use it for adding a route
    /// hint to LN invoices that tells fedimint clients that the invoice can
    /// only be paid internally. Since no LN node with that pb key can exist
    /// other LN senders will know that they cannot pay the invoice.
    fn to_fake_ln_pub_key(&self) -> secp256k1::PublicKey {
        let mut bytes = [0u8; 33];
        bytes[0] = 0x02;
        bytes[1..].copy_from_slice(&<bitcoin_hashes::sha256::Hash as bitcoin_hashes::Hash>::hash(&self.0.to_bytes()[..])[..]);
        secp256k1::PublicKey::from_slice(&bytes).unwrap()
    }
```

Reply:
Asked for help in [bitcoin stack exchange](https://bitcoin.stackexchange.com/questions/118646/derive-secp256k1-curve-public-key-starting-off-from-known-32-bytes-string?noredirect=1#comment135292_118646)

Reply:
Resolved from Pieter's suggestion in that post